osv/ecosystems.py

```diff
@@ -386,6 +396,98 @@ def enumerate_versions(self,
                                        last_affected, limits)
 
 
+class Alpine(Ecosystem):
+  """Alpine packages ecosystem"""
+
+  _APORTS_GIT_URL = 'https://gitlab.alpinelinux.org/alpine/aports.git'
+  _BRANCH_SUFFIX = '-stable'
+  alpine_release_ver: str
+  _GIT_REPO_PATH = 'version_enum/aports/'
+
+  def __init__(self, alpine_release_ver: str):
+    self.alpine_release_ver = alpine_release_ver
+
+  def get_branch_name(self) -> str:
+    return self.alpine_release_ver.lstrip('v') + self._BRANCH_SUFFIX
+
+  def sort_key(self, version):
+    return AlpineLinuxVersion(version)
+
+  @staticmethod
+  def _process_git_log(output: str) -> list:
+    """Takes git log diff output,
+    finds all changes to pkgver and outputs that in an unsorted list"""
+    all_versions = set()
+    lines = [
+        x for x in output.splitlines() if len(x) == 0 or x.startswith('+pkgver')
+    ]
+    # Reverse so that it's in chronological order.
+    # The following loop also expects this order.
+    lines.reverse()
+
+    current_ver = None
+
+    for x in lines:
+      if len(x) == 0:
+        all_versions.add(current_ver)
+        continue
+
+      x_split = x.split('=', 1)
+      if len(x_split) != 2:
+        # Skip the occasional invalid versions
+        continue
+
+      ver = x_split[1]
+      if x.startswith('+pkgver'):
+        current_ver = ver
+        continue
+
+    # Return unsorted list of versions
+    return list(all_versions)
+
+  def enumerate_versions(self,
+                         package: str,
+                         introduced,
+                         fixed=None,
+                         last_affected=None,
+                         limits=None):
+    if work_dir is None:
+      logging.error(
+          "Tried to enumerate alpine version without workdir being set")
+      return []
+
+    get_versions = self._get_versions
+    if shared_cache:
+      get_versions = cached(shared_cache)(get_versions)
+
+    versions = get_versions(self.get_branch_name(), package)
+    self.sort_versions(versions)
+
+    return self._get_affected_versions(versions, introduced, fixed,
+                                       last_affected, limits)
+
+  @staticmethod
+  def _get_versions(branch: str, package: str) -> list[str]:
+    """Get all versions for a package from aports repo"""
+    checkout_dir = os.path.join(work_dir, Alpine._GIT_REPO_PATH)
+    osv.ensure_updated_checkout(
+        Alpine._APORTS_GIT_URL, checkout_dir, branch=branch)
+    directories = glob.glob(
+        '*/' + package.lower() + '/APKBUILD',
+        root_dir=checkout_dir,
```

Comment:
E1123: Unexpected keyword argument 'root_dir' in function call (unexpected-keyword-arg)

Reply:
Resolved in 9caf5fa0f078f5fce53f117b0cdcda48d6cb893c